docs/design/fabric-v2+/fpc-validation.puml

```diff
@@ -97,22 +75,32 @@ end group
     group validation & commit
         ECC1 -> ECC1 : validate block signature
         loop for each tx in block
-            ECC1 -> ECC1 : validate tx (client) signature
-
-            ref over ECC1
-                **Validate endorsement policy**
-                TODO: more details
-                Using EnclaveMSP will transparently verify
-                the following steps:
-                    - is endorsing enclave part of Org_i?
-                    - has enclave a proper attestation?
-                    - is this enclave also registered to
-                      perform execution of this chaincode?
-            end ref
-
-            ECC1 -> ECC1 : read-write check
-
-            ECC1 -> ECC1 : mark tx as valid/invalid
+
+            note over ECC1
+                For every FPC transaction, we first perform
+                default validation and then FPC validation
+            end note
+
+            group default validation
+                ECC1 -> ECC1 : perform default validation
+                ECC1 -> ECC1 : validate tx (client) signature
+                ECC1 -> ECC1 : MVCC check
+                ECC1 -> ECC1 : validate endorsements
+            end
+
+            group fpc-vscc
+                ECC1 -> ECC1 : enclave_id <- extract from transaction
```

Comment:
Don't we need here also first a parsing of the payload into actual fpc-and fpc metadata (signature, enclave id, ..)?  Extraction of id is only one part, you would need extracted signature (so it's not confused with normal endorsement signature).  Ideally, we use the same decomposition/terms as in fpc-cc-invocation ..

Reply:
Updated the flow giving more details and sync with terms used in fpc-cc-invocation